### PR TITLE
Move CI/CD Makefiles configuration to a single place

### PR DESCRIPTION
**Pull request by adrcunha**
Bonuses:
* a few more comments
* document reasons and solutions for auth errors in update-config
* parameterize TestGrid settings

### CODE REVIEW

File: ci/Makefile.base [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 The Knative Authors
```
Comment (chaodaiG) [inline ci/Makefile.base]: 2019

File: ci/Makefile.base [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Default settings for the CI/CD system.
+
+CLUSTER            ?= prow
+PROJECT            ?= knative-tests
+ZONE               ?= us-central1-f
+JOB_NAMESPACE      ?= test-pods
+KNATIVE_CONFIG     ?= config_knative.yaml
+SKIP_CONFIG_BACKUP ?=
+
+CI_DIR           ?= $(realpath $(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
```
Comment (chaodaiG) [inline ci/Makefile.base]: feel like I can use a comment here

File: ci/Makefile.base (hunk 1 above)
Comment (adrcunha) [reply ci/Makefile.base]: Done.

File: ci/Makefile.base (hunk 2 above)
Comment (adrcunha) [reply ci/Makefile.base]: Why now, but not in the original code? Just because of realpath()? :) Done.

Comment (chaodaiG) [reply ci/Makefile.base]: thanks, just think it would be useful for someone who is not super familiar with all of these commands

**Pull request merged**